aten/src/ATen/native/AdaptiveAveragePooling3d.cpp

```diff
@@ -0,0 +1,371 @@
+#include <ATen/ATen.h>
+#include <ATen/NativeFunctions.h>
+
+namespace at {
+namespace native {
+
+namespace {
+
+inline int start_index(int a, int b, int c) {
+  return (int)std::floor((float)(a * c) / b);
+}
+
+inline int end_index(int a, int b, int c) {
+  return (int)std::ceil((float)((a + 1) * c) / b);
+}
+
```

Comment:
Did we lose this comment? Was it inaccurate?

```
/*
 * Description:
 *    This function adaptively average pools an input 5D tensor along dimensions
 *     2, 3 and 4.
 *
 *    gridDim.y blocks work together on a single 2D output plane specified by
 *    (blockIdx.x + offsetZ).
 */
```

Reply:
Hmm, I guess you got a bit confused while reviewing because I found only the below comment in THNN code (C impl).
https://github.com/pytorch/pytorch/blob/cf55670bddf6bb5f6199df2c543c8fb4ca21b08e/aten/src/THNN/generic/VolumetricAdaptiveAveragePooling.c#L10

Also, the new `.cu` file has the comment of description: https://github.com/pytorch/pytorch/pull/19898/files#diff-b9ff310d15f69d0b85cfe7fd87d8f8b9R33